edelweiss_data/api.py

```diff
@@ -572,14 +573,42 @@ def sample(self):
         route = '/datasets/{}/in-progress/sample'.format(self.id)
         return self.api.get(route)
 
+    def update(self, name: typing.Optional[str] = None,
+                     description: typing.Optional[str] = None,
+                     data_source: typing.Optional[dict] = None,
+                     schema: typing.Optional[Schema] = None,
+                     metadata: typing.Optional[dict] = None):
+        '''Update various attributes of a in-progress dataset. All parameters are options; those that are
+        None will not have their values changed.
+
+        :param name: A new name for the dataset
+        :param description: A new description for the dataset
+        :param data_source: A new data_source for the dataset. See set_data_source for a description of a data source.
```

Comment:
`set_data_source` takes a dataset, not a dict as an argument, so perhaps this should be unified in some reasonable way?